python/cudf/cudf/io/parquet.py

```diff
@@ -160,6 +162,113 @@ def read_parquet_metadata(path):
     return num_rows, num_row_groups, col_names
 
 
+def _process_row_groups(paths, fs, filters=None, row_groups=None):
+
+    # Deal with case that the user passed in a directory name
+    file_list = paths
+    if len(paths) == 1 and ioutils.is_directory(paths[0]):
+        paths = ioutils.stringify_pathlike(paths[0])
+
+    # Convert filters to ds.Expression
+    if filters is not None:
+        filters = pq._filters_to_expression(filters)
```

Comment:
Are we OK using a non-public arrow API here? Is there a public alternative, or is it worth vendoring?

Reply:
Good question! This same issue came up in Dask a few times (where we are also using this "private" utility).
@jorisvandenbossche - Is this still the recommended way to translate filters?

Note that this particular line was not actually added in this PR, but it would be good to change it if there is a new "public" API.

Reply:
Although private, it's knowingly being used in several places (such as dask), so we won't be removing it without alternative / deprecation, as with public API. So I think it is fine to use. 
There is an issue about making this public (https://issues.apache.org/jira/browse/ARROW-9672), I can look into that for the coming release, but you will still need to keep using the private one for some time if you want to support older versions.

Reply:
I appreciate the confirmation Joris. I'll keep my eye on that issue in case something changes.  Note that the most ideal solution from the perspective of Dask and RAPIDS is for pyarrow to start recognizing filters specified as the DNF-like list of tuples :)